src/php/lib/Grpc/BaseStub.php

```diff
@@ -72,11 +75,22 @@ public function __construct($hostname, $opts, Channel $channel = null)
                                  'ChannelCredentials::create methods');
         }
         if ($channel) {
-            if (!is_a($channel, 'Grpc\Channel')) {
-                throw new \Exception('The channel argument is not a'.
-                                     'Channel object');
+            if (!is_a($channel, 'Grpc\Channel') &&
+                !is_a($channel, 'Grpc\InterceptorChannel')) {
+                throw new \Exception('The channel argument is not a Channel object '.
+                    'or an InterceptorChannel object created by '.
+                    'Interceptor::intercept($channel, Interceptor|Interceptor[] $interceptors)');
             }
             $this->channel = $channel;
+            while (is_a($channel, 'Grpc\InterceptorChannel')) {
```

Comment:
This entire green block seems excessive. The construction of `InterceptorChannel` should make this invariant true-by-construction.